src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/NewEditCyclePage/EditCyclePage.tsx

```diff
@@ -43,6 +50,7 @@ export function EditCyclePage() {
           form={editCycleForm}
           layout="vertical"
           initialValues={initialFormData}
+          onValuesChange={handleFormValuesChange}
```

Comment:
Is this strong enough to fix toggling? For instance, if a user has 100 reserved tokens, changes it to 0, the back to 100, will the button still be disabled?

Reply:
> Is this strong enough to fix toggling? For instance, if a user has 100 reserved tokens, changes it to 0, the back to 100, will the button still be disabled?

Nup you're right, is this case the button would stay enabled. I planned this initially, but it proved to be a right pain in the A to handle live diff checks in this `onValuesChange`. 

In any case, on second thought, I figured it made sense that if the user has made changes to the form and then back (to no changes), they might not realise everything is actually still the same, and therefore expect the button to work. In this case, we then show them clearly there is no diff in the Review modal, and the launch button from this modal definitely doesn't work if the form has no diff.